lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,109 @@
+      #
+      # @param [Host] host Host to get artifact URL for
+      # @param [Hash] build_details Details of the build in a hash
+      # @param [String] build_url URL to the build
+      #
+      # @return [String] URL to the build artifact
+      def hosts_artifact_url(host, build_details, build_url)
+        platform_key = host[:platform]
+        variant, version, arch, _ = platform_key.to_array
+        # step: map platform strings into platform_key's for the hash
+        if variant == 'debian' && arch == 'x86_64'
+          platform_key.sub!('x86_64', 'amd64')
+        elsif variant == 'fedora'
+          platform_key.sub!(version, "f#{version}")
+        end
+
+        logger.debug("PLATFORM SPECIFIC INFO for #{host} platform '#{platform_key}':")
+        logger.debug(" - #{build_details[:platform_data][platform_key]}, isnil? #{build_details[:platform_data][platform_key].nil?}")
+        artifact_buildserver_path = build_details[:platform_data][platform_key][:artifact]
+        fail_test('no artifact_buildserver_path found') if artifact_buildserver_path.nil?
+        artifact_url = "#{build_url}/#{artifact_buildserver_path}"
+
+        logger.debug("- artifact url: #{artifact_url}")
+        artifact_url_correct = link_exists?( artifact_url )
+        fail_test('artifact url built incorrectly') if !artifact_url_correct
+        artifact_url
+      end
+
+      # install build artifact on the given host
+      #
+      # @param [Host] host Host to install artifact on
+      # @param [String] project_name Name of project for artifact. Needed for OSX installs
+      # @param [String] artifact_url URL of the project install artifact
+      #
+      # @return nil
+      def install_artifact_on(host, artifact_url, project_name)
+        variant, _, _, _ = host[:platform].to_array
+        onhost_package_file = nil
+        if variant == 'solaris'
+          artifact_filename = File.basename( artifact_url )
+          artifact_folder = File.dirname( artifact_url )
+          fetch_http_file( artifact_folder, artifact_filename, '.' )
+          scp_to host, artifact_filename, '/tmp'
+          onhost_package_file = "/tmp/#{ artifact_filename }"
+        elsif variant == 'debian'
+          onhost_package_file = '/tmp/puppet.deb'
+          on host, "wget -O #{ onhost_package_file } #{ artifact_url }"
+        elsif variant == 'osx'
+          on host, "curl -O #{ artifact_url }"
+          onhost_package_file = "#{ project_name }*"
+        end
+
+        if onhost_package_file
+          host.install_local_package( onhost_package_file )
+        else
+          host.install_package( artifact_url )
+        end
+      end
+
+      # install puppet5 project on all hosts
+      #
+      # @param [String] project_name Name of the project to install
+      # @param [String] project_sha SHA ref of the project to install
+      #
+      # @return nil
+      def puppet5_install(project_name, project_sha)
```

Comment:
Does this mean we need to change our helper invocations to `puppet5_install` in order to use puppet 5 instead of puppet 4?